tests/ha/single_pod_restart/test_data_server_pod_restart.py

```diff
@@ -518,3 +524,450 @@ def test_io_data_server_pod_restart(self):
+        LOGGER.info("STEP 10.1: Performed READs/Verify on data written in healthy mode in step-1")
+
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("STEP 10.2: Perform READs/Verify on data written in degraded mode when "
+                        "data pod down in step-4")
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                        log_prefix=self.test_prefix_deg,
+                                                        skipwrite=True, skipcleanup=True,
+                                                        setup_s3bench=False, nsamples=5, nclients=5)
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("STEP 10.2: Performed READs/Verify on data written in degraded mode when"
+                        " data pod down in step-4")
+
+            LOGGER.info("STEP 10.3: Perform READs/Verify on data written after data pod restart "
+                        "in step-8")
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                        log_prefix=self.test_prefix_deg,
+                                                        skipwrite=True, skipcleanup=True,
+                                                        setup_s3bench=False, nsamples=5, nclients=5)
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("STEP 10.3: Performed READs/Verify on data written after data pod restart "
+                        "in step-8")
+
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("Step 11: Create new bucket and perform WRITEs/READs/Verify with variable "
+                        "object sizes when server pod down")
+            self.test_prefix_server_deg = f'test-45529-server_deg-{int(perf_counter_ns())}'
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                        log_prefix=self.test_prefix_server_deg,
+                                                        skipcleanup=True, setup_s3bench=False,
+                                                        nsamples=5, nclients=5)
+            assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 11: Create new objects and perform WRITEs/READs/Verify with variable "
+                    "object sizes when server pod down")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                    log_prefix=self.test_prefix,
+                                                    skipcleanup=True, setup_s3bench=False,
+                                                    nsamples=5, nclients=5)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 11: Performed WRITEs/READs/Verify with variable object sizes "
+                    "when server pod down")
+
```

Comment:
was this test we decided to run IOs during server pod restart?

Reply:
The next one we decide to do that. TEST-45536